pkg/cluster/prepare.go

```diff
@@ -0,0 +1,132 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/fluxcd/pkg/untar"
+	"github.com/weaveworks/pctl/pkg/git"
+)
+
+const (
+	releaseUrl = "https://github.com/weaveworks/profiles/releases"
+	kubectlCmd = "kubectl"
+)
+
+// Fetcher will download a manifest tar file from a remote repository.
+type Fetcher struct {
+	Client *http.Client
+}
+
+// Applier applies the previously generated manifest files.
+type Applier struct {
+	Runner git.Runner
+}
+
+// Preparer will prepare an environment.
+type Preparer struct {
+	PrepConfig
+	Fetcher *Fetcher
+	Applier *Applier
+}
+
+// PrepConfig defines configuration options for prepare.
+type PrepConfig struct {
+	// BaseURL is given even one would like to download manifests from a fork
+	// or a test repo.
+	BaseURL     string
+	Location    string
+	Version     string
+	KubeContext string
+	KubeConfig  string
+	DryRun      bool
+}
+
+// NewPreparer creates a preparer with set dependencies ready to be used.
+func NewPreparer(cfg PrepConfig) *Preparer {
+	if cfg.BaseURL == "" {
+		cfg.BaseURL = releaseUrl
+	}
+	return &Preparer{
+		PrepConfig: cfg,
+		Fetcher: &Fetcher{
+			Client: http.DefaultClient,
+		},
+		Applier: &Applier{
+			Runner: &git.CLIRunner{},
```

Comment:
do we want to move this runner to its own lil package, rather than importing the git one (bit misleading)?